tests/ManageCourses.Tests/DbIntegration/EnrichmentServiceTests.cs

```diff
@@ -0,0 +1,202 @@
+            const string trainWithDisabilityUpdatedText = "TrainWithDisabilily Text updated";
+            const string trainWithUsUpdatedText = "TrainWithUs Text updated";
+            const string instDesc = "school1 description enrichement";
+
+        var enrichmentService = new EnrichmentService(Context);
+            var model = new UcasInstitutionEnrichmentPostModel
+            {
+                EnrichmentModel = new InstitutionEnrichmentModel
+                {
+                    TrainWithDisability = trainWithDisabilityText,
+                    TrainWithUs = trainWithUsText,
+                    AccreditingProviderEnrichments = new List<AccreditingProviderEnrichment>
+                    {
+                        new AccreditingProviderEnrichment
+                        {
+                            UcasInstitutionCode = _accreditingInstCode,
+                            Description = instDesc,
+                        }
+                    }
+                }
+            };
+            //test save
+            enrichmentService.SaveInstitutionEnrichment(model, _providerInstCode, _email);
+            //test get
+            var result = enrichmentService.GetInstitutionEnrichment(_providerInstCode, _email);
+
+            result.Should().NotBeNull();
+            result.EnrichmentModel.Should().NotBeNull();
+            result.EnrichmentModel.TrainWithDisability.Should().BeEquivalentTo(trainWithDisabilityText);
+            result.EnrichmentModel.TrainWithUs.Should().BeEquivalentTo(trainWithUsText);
+            
+            //test update
+            var updatedmodel = new UcasInstitutionEnrichmentPostModel
+            {
+                EnrichmentModel = new InstitutionEnrichmentModel
+                {
+                    TrainWithDisability = trainWithDisabilityUpdatedText,
+                    TrainWithUs = trainWithUsUpdatedText,
+                    AccreditingProviderEnrichments = new List<AccreditingProviderEnrichment>
+                    {
+                        new AccreditingProviderEnrichment
+                        {
+                            UcasInstitutionCode = _accreditingInstCode,
+                            Description = instDesc,
+                        }
+                    }
+                }
+            };
+
+            enrichmentService.SaveInstitutionEnrichment(updatedmodel, _providerInstCode, _email);
+            var updateResult = enrichmentService.GetInstitutionEnrichment(_providerInstCode, _email);
+            updateResult.EnrichmentModel.TrainWithDisability.Should().BeEquivalentTo(trainWithDisabilityUpdatedText);
+            updateResult.EnrichmentModel.TrainWithUs.Should().BeEquivalentTo(trainWithUsUpdatedText);
+        }
+        [Test]
+        [TestCase("", "")]
+        [TestCase(null, null)]
+        [TestCase("eqweqw", "qweqweq")]
+        public void Test_SaveInstitutionEnrichment_should_error(string instCode, string email)
```

Comment:
Whos exception are you testing?
Why are you testing theses exception?
If the point is because of bad argument, then its bad argument exception and protect the uneeded use of db to flush out the exception